pkg/proxy/ipvs/meta_proxier.go

```diff
@@ -153,25 +153,30 @@ func (proxier *metaProxier) OnEndpointsSynced() {
 // OnEndpointSliceAdd is called whenever creation of a new endpoint slice object
 // is observed.
 func (proxier *metaProxier) OnEndpointSliceAdd(endpointSlice *discovery.EndpointSlice) {
-	// noop
+	proxier.ipv4Proxier.OnEndpointSliceAdd(endpointSlice)
```

Comment:
Does the IP family need to be checked first similar to the `OnEndpoints*` methods?

Reply:
Even if EndpointSlice can contain address in both families should we still check if it contains only v4 or only v6 first?

Reply:
I think this might be over optimization, giving that a slice can grow to a 1000 entry we may actually get opposite results. I understand that you are trying to avoid the call to `syncProxyRules()` Let us talk tom, i want to understand more.